examples/register_and_change_username.php

```diff
@@ -43,35 +41,17 @@
     ],
 ];
 
-$dispatch = Kernel\buildCommandDispatcher(
-    $commandMap,
-    $factories['eventStore'],
-    $factories['snapshotStore']
-);
+$dispatch = Kernel\buildCommandDispatcher($factories['eventStore']())($factories['snapshotStore']())($commandMap);
```

Comment:
Is a SnapshotStore always required or do I have to call that build step with "null"?
What is the advantage of "one argument per function" instead of just a buildCommandDispatcher with two mandatory and one optional argument?